modules/custom-status/lib/custom-status-block.js

```diff
@@ -1,112 +1,211 @@
-		} );
-	}
+const pluginName = 'vip-workflow-custom-status';
+const sidebarName = 'vip-workflow-sidebar';
+
+// Plugin sidebar
+const CustomSaveButtonSidebar = ( {
+	postType,
+	savedStatus,
+	editedStatus,
+	isUnsavedPost,
+	isSavingPost,
+	onUpdateStatus,
+} ) => {
+	const isTinyViewport = useViewportMatch( 'small', '<' );
+	const isWideViewport = useViewportMatch( 'wide', '>=' );
+
+	const isCustomSaveButtonVisible = useMemo(
+		() => isCustomSaveButtonEnabled( isUnsavedPost, postType, savedStatus ),
+		[ isUnsavedPost, postType, savedStatus ]
+	);
 
-	const selectedStatus = select( 'core/editor' ).getEditedPostAttribute( 'status' );
-	// check if the post status is in the list of custom statuses, and only then issue the notices
-	if (
-		typeof vw_publish_guard_enabled !== 'undefined' &&
-		vw_publish_guard_enabled &&
-		statuses.find( status => status.value === selectedStatus )
-	) {
-		const has_publish_capability =
-			select( 'core/editor' ).getCurrentPost()?._links?.[ 'wp:action-publish' ] ?? false;
-		if ( postLocked && has_publish_capability ) {
-			postLocked = false;
-			dispatch( 'core/notices' ).removeNotice( 'publish-guard-lock' );
-		} else if ( ! postLocked && ! has_publish_capability ) {
-			postLocked = true;
-			dispatch( 'core/notices' ).createInfoNotice( __( 'This post is locked from publishing.' ), {
-				id: 'publish-guard-lock',
-				type: 'snackbar',
-			} );
+	const isCustomSaveButtonDisabled = isSavingPost;
+
+	// Selectively disable the native save button when workflow statuses are in effect
+	useEffect( () => {
+		const editor = document.querySelector( '#editor' );
+
+		if ( isCustomSaveButtonVisible ) {
+			editor.classList.add( 'disable-native-save-button' );
+		} else {
+			editor.classList.remove( 'disable-native-save-button' );
 		}
-	}
-} );
+	}, [ isCustomSaveButtonVisible ] );
+
+	const nextStatusTerm = useMemo( () => getNextStatusTerm( savedStatus ), [ savedStatus ] );
+
+	useInterceptPluginSidebar(
+		`${ pluginName }/${ sidebarName }`,
+		( _isSidebarActive, _toggleSidebar ) => {
+			if ( isCustomSaveButtonDisabled ) {
+				return;
+			}
+
+			if ( nextStatusTerm ) {
+				onUpdateStatus( nextStatusTerm.slug );
+				dispatch( editorStore ).savePost();
+			}
+		}
+	);
+
+	const buttonText = getCustomSaveButtonText( nextStatusTerm, isWideViewport );
+
+	const InnerSaveButton = (
+		<CustomInnerSaveButton
+			buttonText={ buttonText }
+			isSavingPost={ isSavingPost }
+			isDisabled={ isCustomSaveButtonDisabled }
+			isTinyViewport={ isTinyViewport }
+		/>
+	);
+
+	return (
+		<>
+			{ /* "Extended Post Status" in the sidebar */ }
+			<CustomStatusSidebar
+				postType={ postType }
+				status={ editedStatus }
+				onUpdateStatus={ onUpdateStatus }
+			/>
+
+			{ /* Custom save button in the toolbar */ }
+			{ isCustomSaveButtonVisible && (
+				<PluginSidebar name={ sidebarName } title={ buttonText } icon={ InnerSaveButton }>
+					{ /* Use this space to show approve/reject UI or other sidebar controls */ }
```

Comment:
Would switch this to a `//ToDo` structure. Trying to standardize so it's easy to find these ToDo entry points

Reply:
Fixed in https://github.com/Automattic/vip-workflow-plugin/pull/24/commits/240034cef39a73fa014a1ade855fa48702368294.